codes/dataops/opencv_transforms/README.md

```diff
@@ -5,81 +5,90 @@ This is intended as a faster drop-in replacement of Pytorch's Torchvision augmen
 This repository is the result of merging [jbohnslav](https://github.com/jbohnslav/opencv_transforms) and [YU-Zhiyang](https://github.com/YU-Zhiyang/opencv_transforms_torchvision) repositories which had the same purpose, and my own OpenCV-based augmentations from [BasicSR](https://github.com/victorca25/BasicSR), in order to allow to refactor the project's data flow and streamline to use the Torchvision's API as a standard. This enables changing or combining different base frameworks (OpenCV, Pillow/Pillow-SIMD, etc) only by modifying the imported library and also to easily switch to other replacements like [Kornia](https://github.com/kornia/kornia), [Albumentations](https://github.com/albumentations-team/albumentations), or [Rising](https://github.com/PhoenixDL/rising), based on the user's needs.
 
 Most functions in Pytorch transforms are reimplemented, but there are some considerations:
-   1) ToPILImage is not implemented, we use OpenCV instead (ToCVImage). However, the original ToPILImage in ~transforms can be used to save the tensor as a PIL image if required. Once transformed into tensor format, images have RGB channel order in both cases. 
-   2) OpenCV images are Numpy arrays. OpencV supports uint8, int8, uint16, int16, int32, float32, float64. Certain operations (like `cv.CvtColor()`) do require to convert the arrays to OpenCV type (with `cv.fromarray()`).
-   3) The affine transform in the original one only has 5 degrees of freedom, YU-Zhiyang implemented an Affine transform with 6
-    degress of freedom called `RandomAffine6` (can be found in [transforms.py](opencv_transforms/transforms.py)). The
-     original method `RandomAffine` is also available and reimplemented with OpenCV.
-   4) The rotate function is clockwise, however the original one is anticlockwise.
-   5) Some new augmentations have been added, in comparison to Torchvision's and are indicated in **Support** with and asterisk.
-   6) **The outputs of the OpenCV versions are almost the same as the original one's (it's possible to test by running [test.py](/test.py)) directly with test images**.
-
-## Support:
+
```

Comment:
Please send a PR to the https://github.com/victorca25/opencv_transforms repo instead, I can synchronize the changes from there back into BasiSR

Reply:
Oh nice repo, if we gonna do that, could we use submodules on the repo? That way you can do `git clone --recurse-submodules https://github.com/victorca25/BasicSR` to always have an up-to-date opencv_transforms?

Regardless, pull request made.

Reply:
using a submodule is definitely a better option than having a copy of it here, though submodules will cause an issue for anyone that downloads the repository as a zip

Reply:
It's not a copy, it's a subtree. A subtree was cleaner for most people to use, submodule adds more commands for the final user who may not know how it works.

Reply:
That makes sense then